### PR TITLE
Document some particularities of the revoke subcommand

### PR DESCRIPTION
**Pull request by pde**
Give revoke better cli help. Also hint at the weirdness of #3922.

### CODE REVIEW

File: certbot/cli.py [hunk 1]
```diff
@@ -912,8 +913,9 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
         help="Silence all output except errors. Useful for automation via cron."
              " Implies --non-interactive.")
     # overwrites server, handled in HelpfulArgumentParser.parse_args()
-    helpful.add("testing", "--test-cert", "--staging", action='store_true', dest='staging',
-        help='Use the staging server to obtain test (invalid) certs; equivalent'
+    helpful.add(["testing", "revoke"], "--test-cert", "--staging",
```
Comment (ohemorange) [inline certbot/cli.py]: If we're adding this to "revoke", should it also be in "certonly", "run", "renew", etc?

Comment (pde) [inline certbot/cli.py]: It's a matter of judiciously selecting the most relevant options to be documented in those topics; `certonly` already has `--dry-run` which is probably better for most use cases, but I can add `--test-cert` to run.

**Comment by pde** [comment]
The error from `revoke` looks like this:
```
--config-dir /tmp/xyz/ revoke --cert-path /tmp/xyz/live/isnot.org/fullchain.pem --agree-t --regi
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Registering without email!
An unexpected error occurred:
The request message was malformed :: No such certificate
Please see the logfiles in /var/log/letsencrypt for more details.

IMPORTANT NOTES:
 - Your account credentials have been saved in your Certbot
   configuration directory at /tmp/xyz. You should make a secure
   backup of this folder now. This configuration directory will also
   contain certificates and private keys obtained by Certbot so making
   regular backups of this folder is ideal.
```
Note the creation of a production account in what was previously a staging-only setup.  If you already have a production account it looks like this:

```
certbot --config-dir /tmp/xyz/ revoke --cert-path /tmp/xyz/live/isnot.org/fullchain.pem
Saving debug log to /var/log/letsencrypt/letsencrypt.log
An unexpected error occurred:
The request message was malformed :: No such certificate
Please see the logfiles in /var/log/letsencrypt for more details.
```
The `No such certificate` error is generated by boulder, so we can't directly change it. For the record I don't think this is a true fix for #3922 (which would really be to examine the `Issuer` of a cert to set a sensible default revocation `server` value).

**Comment by ohemorange** [comment]
Do all these commits belong in this PR?

**Comment by pde** [comment]
I was wondering where those commits had gone :).  Rebased/fixed.

**Comment by ohemorange** [comment]
travis?

**Pull request merged**